### PR TITLE
(maint) add subcommand specs for options passthrough

### PR DESCRIPTION
**Pull request by tvpartytonight**

### CODE REVIEW

File: spec/beaker/subcommand_spec.rb [hunk 1]
```diff
@@ -32,6 +32,72 @@ module Beaker
       end
     end
 
+    context 'ensure that beaker options can be passed through' do
+
+      BEAKER_OPTIONS_LIST = [
```
Comment (ericwilliamson) [inline spec/beaker/subcommand_spec.rb]: any reason this is a constant instead of a `let(:)`?

File: spec/beaker/subcommand_spec.rb [hunk 2]
```diff
@@ -32,6 +32,72 @@ module Beaker
...
+        'exec_manual_tests',
+        'test-tag-exclude',
+        'test-tag-and',
+        'test-tag-or',
+        'xml',
+        'type',
+        'debug',
+      ]
+
+      let( :yaml_store_mock ) { double('yaml_store_mock') }
```
Comment (ericwilliamson) [inline spec/beaker/subcommand_spec.rb]: style nitpick: remove spaces around parents `let(:yaml_store_mock)`

File: spec/beaker/subcommand_spec.rb [hunk 3]
```diff
@@ -32,6 +32,72 @@ module Beaker
...
+          allow(YAML::Store).to receive(:new).with(SubcommandUtil::SUBCOMMAND_STATE).and_return(yaml_store_mock)
+          allow(yaml_store_mock).to receive(:transaction).and_yield
+          allow(yaml_store_mock).to receive(:[]=).with('provisioned', false)
+          allow(File).to receive(:open)
+          allow_any_instance_of(Beaker::Logger).to receive(:notify).twice
+          expect {Beaker::Subcommand.start(['init', '--hosts', 'centos', "--#{option}"])}.to_not output(/ERROR/).to_stderr
+        end
+      end
+
+      it " should error with a bad option here " do
```
Comment (ericwilliamson) [inline spec/beaker/subcommand_spec.rb]: style nitpick: remove space after/before quotes

File: spec/beaker/subcommand_spec.rb (hunk 1 above)
Comment (tvpartytonight) [reply spec/beaker/subcommand_spec.rb]: I started this thinking this list would be useful for more than just this test, but I wound up deciding it should be limited to just this test for now. I"ll change it.

File: spec/beaker/subcommand_spec.rb (hunk 2 above)
Comment (tvpartytonight) [reply spec/beaker/subcommand_spec.rb]: I want to be consistent with the rest of this file, so... 🤷‍♂️

Comment (ericwilliamson) [reply spec/beaker/subcommand_spec.rb]: ah didn't notice the rest of the file was doing that...oh well

**Comment by tvpartytonight** [comment]
@ericwilliamson ready for merge?

**Pull request merged**